crates/router/src/connector/cybersource/transformers.rs

```diff
@@ -85,15 +94,20 @@ pub struct CybersourceZeroMandateRequest {
     client_reference_information: ClientReferenceInformation,
 }
 
-impl TryFrom<&types::SetupMandateRouterData> for CybersourceZeroMandateRequest {
+impl TryFrom<&CybersourceRouterData<&types::SetupMandateRouterData>>
+    for CybersourceZeroMandateRequest
+{
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &types::SetupMandateRouterData) -> Result<Self, Self::Error> {
+    fn try_from(
+        item_data: &CybersourceRouterData<&types::SetupMandateRouterData>,
+    ) -> Result<Self, Self::Error> {
+        let item = item_data.router_data.clone();
         let email = item.get_billing_email().or(item.request.get_email())?;
         let bill_to = build_bill_to(item.get_optional_billing(), email)?;
 
         let order_information = OrderInformationWithBill {
             amount_details: Amount {
-                total_amount: "0".to_string(),
+                total_amount: item_data.amount.clone(),
                 currency: item.request.currency,
             },
             bill_to: Some(bill_to),
```

Comment:
These changes are not required. Hard code the value to zero.
`StringMajorUnit("0".to_string())`

Reply:
You can make the function public

Reply:
sure making the new function public